src/controller.coffee

```diff
@@ -3672,6 +3690,16 @@ define ['droplet-helper',
 
     @redrawHighlights()
 
+  Editor::markBlock = (block, style) ->
+    key = @nextMarkedBlockId++
+
+    @markedBlocks[key] = {
+      model: block
+      style: style
+    }
```

Comment:
The key is for later removal; it's an object because we want to be able to do things like `delete @markedBlocks[1]` and still have `2` point to the correct mark at `@markedBlocks[2]`. We could do this with an array, too, but an object seemed cleaner.